src/triggers.js

```diff
@@ -133,7 +133,9 @@ function notifySidebars() {
 
 function printLabel(order_id) {
   var worksheet = new Worksheet();
-  worksheet.validateAndAdvanceState(order_id,'Present');
+  if (worksheet.validateAndAdvanceState(order_id,'Present') === -1) {
```

Comment:
My thought is that we should print the label first; only if that is successful should we change the state to 'Labeled'.